src/components/AuthContainer.js

```diff
@@ -26,8 +26,7 @@ const AuthContainer = ({ children }) => {
     [seenAzurePreview === false && isAzureUser(), () => h(AzurePreview)],
     [registrationStatus === undefined && !isPublic, authspinner],
     [registrationStatus === userStatus.unregistered, () => h(Register)],
-    [userNeedsToAcceptTos === undefined && !isPublic, authspinner],
-    [userNeedsToAcceptTos && name !== 'privacy', () => h(TermsOfService)],
+    [displayTosPage && _.isUndefined(termsOfService.userContinuedUnderGracePeriod) && name !== 'privacy', () => h(TermsOfService)],
```

Comment:
On second look, this is the only place `displayTosPage` is used, so why isn't the check for `_.isUndefined(termsOfService.userContinuedUnderGracePeriod)` part of the logic for setting `displayTosPage`? As is, the `displayTosPage` variable is only part of the logic for displaying the ToS page.